hibernate-core/src/main/java/org/hibernate/hql/spi/id/cte/CteValuesListBulkIdStrategy.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.hql.spi.id.cte;
+
+import org.hibernate.boot.spi.MetadataImplementor;
+import org.hibernate.boot.spi.SessionFactoryOptions;
+import org.hibernate.engine.jdbc.connections.spi.JdbcConnectionAccess;
+import org.hibernate.engine.jdbc.spi.JdbcServices;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.hql.internal.ast.HqlSqlWalker;
+import org.hibernate.hql.spi.id.MultiTableBulkIdStrategy;
+
+/**
+ * This bulk-id strategy uses a CTE with a VALUE list to hold the identifiers,
+ * which are later used by the update or delete statement:
+ *
+ * <code>
+ * with HT_Person (id ) as (
+ *     select
+ *         id
+ *     from (
+ *         values
+ *             (?),
+ *             (?),
+ *             (?)
+ *             (?)
+ *     ) as HT (id)
+ * )
+ * delete
+ * from
+ *     Person
+ * where
+ *     ( id ) in (
+ *         select
+ *             id
+ *         from
+ *             HT_Person
+ *     )
+ * </code>
```

Comment:
I didn't notice it. Thanks for the hint. I changed it to using `<pre>` tags, and it looks great now.